docs/edgeql/path_resolution.rst

```diff
@@ -0,0 +1,177 @@
+
+In some cases, this works out fine, but in others it doesn't make sense. Take
+this example:
+
+.. code-block:: edgeql
+
+    select User.first_name ++ ' ' ++ User.last_name;
+
+Should the result of this query be every ``first_name`` in your ``User``
+objects concatenated with every ``last_name``? That's probably not what you
+want.
+
+This is why, in cases where multiple element-wise arguments share a common path
+(``User.`` in this example), EdgeDB factors out the common path rather than
+using cartesian multiplication.
+
+.. code-block:: edgeql-repl
+
+    db> select User.first_name ++ ' ' ++ User.last_name;
+    {'Mina Murray', 'Jonathan Harker', 'Lucy Westenra', 'John Seward'}
+
+We assume this is what you want, but if your goal is to get the cartesian
+product, you can accomplish it one of three ways. You could use
+:eql:op:`detached`.
+
+.. code-block:: edgeql-repl
+
+    edgedb> select User.first_name ++ ' ' ++ detached User.last_name;
+    {
+      'Mina Murray',
+      'Mina Harker',
+      'Mina Westenra',
+      'Mina Seward',
+      'Jonathan Murray',
+      'Jonathan Harker',
+      'Jonathan Westenra',
+      'Jonathan Seward',
+      'Lucy Murray',
+      'Lucy Harker',
+      'Lucy Westenra',
+      'Lucy Seward',
+      'John Murray',
+      'John Harker',
+      'John Westenra',
+      'John Seward',
+    }
+
+You could use :ref:`with <ref_eql_with>` to attach a different symbol to
+your set of ``User`` objects.
+
+.. code-block:: edgeql-repl
+
+    edgedb> with U := (select User)
+    ....... select U.first_name ++ ' ' ++ User.last_name;
+    {
+      'Mina Murray',
+      'Mina Harker',
+      'Mina Westenra',
+      'Mina Seward',
+      'Jonathan Murray',
+      'Jonathan Harker',
+      'Jonathan Westenra',
+      'Jonathan Seward',
+      'Lucy Murray',
+      'Lucy Harker',
+      'Lucy Westenra',
+      'Lucy Seward',
+      'John Murray',
+      'John Harker',
+      'John Westenra',
+      'John Seward',
+    }
+
+Or you could leverage the effect scopes have on path resolution. More on that
+:ref:`in the Scopes section <ref_eql_path_resolution_scopes>`.
+
+The reason ``with`` works here even though the alias ``U`` refers to the exact
+same set is that we only assume you want the path factored in this way when you
+use the same *symbol* to refer to a set. This means operations with
+``User.first_name`` and ``User.last_name`` *do* get the common path factored
+while ``U.first_name`` and ``User.last_name`` *do not* and are resolved with
+cartesian multiplication.
+
+That may leave you still wondering why ``U`` and ``User`` did not get a common
+path factored. ``U`` is just an alias of ``select User`` and ``User`` is the
+same symbol that we use in our name query. That's true, but EdgeDB doesn't
+factor in this case because of the queries' scopes.
+
+.. _ref_eql_path_resolution_scopes:
+
+Scopes
+------
+
+Scopes change the was path resolution works. Two sibling select queries — that
```

Comment:
`Scopes change the was path resolution works.` Not sure how to read this sentence. Could you clarify?

Reply:
That's a typo. Thanks for catching it! Just pushed a fix. Should be `way` instead of `was`.